trade-module/src/main/java/io/edpn/backend/trade/adapter/persistence/repository/MybatisLocateCommodityRepository.java

```diff
@@ -17,26 +17,86 @@ public interface MybatisLocateCommodityRepository {
+            						buy_price,
+            						sell_price,
+            						x_coordinate,
+            						y_coordinate,
+            						z_coordinate,
+            						planetary,
+            						require_odyssey,
+            						fleet_carrier
+            				 FROM filtered_lcv
+            				 WHERE 1 = 1
+            					<if test='!(includePlanetary == null || includePlanetary)'>AND planetary = false </if>
+            					<if test='!(includeOdyssey == null || includeOdyssey)'>AND require_odyssey = false </if>
+            					<if test='!(includeFleetCarriers == null || includeFleetCarriers)'>AND fleet_carrier = false </if>
+            					<if test='minSupply != null'>AND stock >= #{minSupply} </if>
+            					<if test='minDemand != null'>AND demand >= #{minDemand} </if>
+            					<choose>
+            						<when test='shipSize == "LARGE"'>
+            							AND max_landing_pad_size = 'LARGE'
+            						</when>
+            						<when test='shipSize == "MEDIUM"'>
+            							AND max_landing_pad_size IN ('MEDIUM', 'LARGE')\s
+            						</when>
+            						<when test='shipSize == "SMALL"'>
+            							AND max_landing_pad_size IN ('MEDIUM', 'LARGE', 'SMALL', 'UNKNOWN')
+            						</when>
+            					</choose>
+            					)
+            	SELECT cte.timestamp,
+            		   cte.commodity_id,
+            		   cte.station_id,
+            		   cte.system_id,
+            		   cte.stock,
+            		   cte.demand,
+            		   cte.buy_price,
+            		   cte.sell_price,
+            		   <choose>
+            			<when test='xCoordinate == null || yCoordinate == null || zCoordinate == null'>
+            				sqrt(pow(cte.x_coordinate, 2) +
+            				pow(cte.y_coordinate, 2) +
+            				pow(z_coordinate, 2)) as distance,
+            			</when>
+            			<otherwise>
+            				sqrt(pow(#{xCoordinate} - cte.x_coordinate, 2) +
+            				pow(#{yCoordinate} - cte.y_coordinate, 2) +
+            				pow(#{zCoordinate} - z_coordinate, 2)) as distance,
+            			</otherwise>
+            		   </choose>
+            		   cte.planetary,
+            		   cte.require_odyssey,
+                		   cte.fleet_carrier,
+            		   <if test='page != null'>
+                           #{page.size} as page_size,
+                           #{page.page} as page_number,
+                           (SELECT count(1) FROM cte) as total_items
+            		   </if>
+            		   <if test='page == null'>
+                           null as page_size,
+                           null as current_page,
+                           null as total_items
+            		   </if>
+            	FROM cte
+            	ORDER BY distance
+            	<if test='page != null'>
+            		LIMIT #{page.size}
+            		OFFSET (#{page.size} * #{page.page})
+            	</if>
+            </script>
+            """
```

Comment:
I honestly don't get why annotations with embedded scripting generating SQL would be a good idea. Personally, would choose a type-safe programmatic approach any given day. JOOQ or QueryDSL or something.

Please, at least tell me the validity and correctness of the scripts are checked at compile time!

Reply:
Disclaimer: I don't have any prior experience on MyBatis. In fact, been _trying_ to read up on it, but their own documentation seems a bit... thin. Could use a few pointers or a reading list.